src/App.tsx

```diff
@@ -1,24 +1,66 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './components/Clock';
 
-const App: React.FC = () => {
-  window.setInterval(() => {
-    const date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
+type State = {
+  clockName: number,
+  isClockVisible: boolean,
 };
 
+class App extends React.Component<{}, State> {
+  state = {
+    clockName: 0,
+    isClockVisible: true,
+    textButt: 'Hide',
+  };
+
+  componentDidUpdate(_prevProps: Readonly<State>, prevState: Readonly<State>) {
+    // eslint-disable-next-line no-console
+    console.log(`The Clock was renamed from ${prevState.clockName} to ${this.state.clockName}`);
+  }
+
+  randomName() {
+    this.setState({
+      clockName: Math.floor(Math.random() * 1000),
+    });
+  }
+
+  showClock() {
```

Comment:
функція повинна робити те що вказано в її імені. showClock повинна показувати годинни а не показувати і ховати
```suggestion
  toggleClockView() {
```